gemrb/core/DisplayMessage.cpp

```diff
@@ -193,10 +193,11 @@ std::map<GUIColors, std::string> DisplayMessage::GetAllColors() const
 	std::map<GUIColors, std::string> auxiliaryColors;
 	AutoTable colorTable = gamedata->LoadTable("colors", true);
 	assert(colorTable);
-	for (TableMgr::index_t r = 0; r < colorTable->GetRowCount(); r++) {
-		if (static_cast<int>(r) >= static_cast<int>(GUIColors::WHITE) && static_cast<int>(r) <= static_cast<int>(GUIColors::DIALOGPARTY)) {
-			auxiliaryColors[static_cast<GUIColors>(r)] = colorTable->GetRowName(r);
-		}
+	TableMgr::index_t index = static_cast<int>(GUIColors::FIRST_COLOR) + 1;
```

Comment:
If you bump FIRST_COLOR to 7, then you could directly cast both index and finish to TableMgr::index_t. Or not cast them at all and have them be of GUIColors type (not sure if that would work below), so you'd only have to have one cast to index_t when looking up the row name.

Reply:
> If you bump FIRST_COLOR to 7, then you could directly cast both index and finish to TableMgr::index_t. Or not cast them at all and have them be of GUIColors type (not sure if that would work below), so you'd only have to have one cast to index_t when looking up the row name.

The problem to bump it is that it will mess up the colors numbers, I tried using them without casting and I can't do operation on them. I could store the FIRST_COLOR and don't do any operation, but seams like a waste of memory.

Reply:
enums can have duplicate values without issues, you just have to explicitly specify them, so the +1 rule doesn't take over.